Features/44 internal migration

crangelsmith opened the pull request:
Closes #44 

- [x] Implementing internal migration module
- [x] Get a requirement for minimum amount of time from last migration
- [x] Start notebook pipeline  

Extra implementation:

- [x] Implement internal migration assignment functions based on dummy matrices
- [x] Use OD matrices to compute real rates (@kasra-hosseini)

kasra-hosseini commented on tests/population/test_internal_migration.py:
```diff
@@ -0,0 +1,113 @@
+    filename_internal_outmigration_name = 'InternalOutmig2011_LEEDS2.csv'
+    path_to_internal_outmigration_file = "{}/{}".format(path_dir, filename_internal_outmigration_name)
+
+    filename_total_population = 'MY2011AGEN.csv'
```
I am not sure this file is needed?

kasra-hosseini commented:
@crangelsmith I added the OD matrix to the internal migration component. Also, updated the test.

crangelsmith commented:
I removed the large OD matrix file and modified the test by adding a line that downloads the file before running (currently this is just a link from google drive but we can improve it once we have the new matrices). This allows the CI to run without the file being on the repo. I'll merge this PR and will start a fresh one once we have the new matrices from Luke. 
@kasra-hosseini

kasra-hosseini commented:
Sounds good, thanks!
